packages/sshnoports/bundles/shell/install.sh

```diff
@@ -0,0 +1,296 @@
+  unit_name="$1"
+  no_mac
+  dest="$systemd_dir/$unit_name"
+  cp "$script_dir/systemd/$unit_name" "$dest"
+  post_systemd_message
+}
+
+install_systemd_sshnpd() {
+  root_only
+  install_single_binary "sshnpd"
+  install_single_binary "sshrv"
+  install_systemd_unit "sshnpd.service"
+}
+
+install_systemd_sshrvd() {
+  root_only
+  install_single_binary "sshrvd"
+  install_systemd_unit "sshrvd.service"
+}
+
+systemd() {
+  case "$1" in
+    --help) usage; exit 0;;
+    sshnpd) install_systemd_sshnpd;;
+    sshrvd) install_systemd_sshrvd;;
+    *)
+      echo "Unknown systemd unit: $1";
+      usage;
+      exit 1;
+  esac
+}
+
+# HEADLESS SERVICES #
+
+post_headless_message() {
+  echo "Headless job installed, make sure to configure the job by editing $dest"
+  echo "Learn more in $script_dir/headless/README.md"
+  echo ""
+  echo "The job will start upon next system boot"
+  echo "To start the job immediately:"
+  echo "  $command &"
+  echo ""
+  echo "Warning: logs stored at $log_file and $err_file have unlimited potential size"
+  echo "In a production environment, it's recommended that you install via systemd"
+  echo "or use a cron job to call logrotate on these log files if systemd is not available"
+}
+
+install_headless_job() {
+  job_name=$1
+  dest="$user_bin_dir/$job_name"
+  cp "$script_dir/headless/$job_name" "$dest"
+
+  log_file="$user_sshnpd_dir/logs/$job_name.log"
+  err_file="$user_sshnpd_dir/logs/$job_name.err"
+
+  command="nohup $dest > $log_file 2> $err_file"
+  cron_entry="@reboot $command"
+  crontab_contents=$(crontab -l 2>/dev/null)
+
+  has_existing_entry=$(echo "$crontab_contents" | grep -Fxq "$cron_entry")
+  if $has_existing_entry; then
+    echo "cron job already installed, killing old $job_name process"
+    pgrep "$command" | xargs kill
+  else
+    echo "Installing cron job: $cron_entry"
+    (echo "$crontab_contents"; echo "$cron_entry") | crontab -;
+  fi
+
+  echo ""
+  post_headless_message
+}
+
+install_headless_sshnpd() {
+  install_single_binary "sshnpd"
+  install_single_binary "sshrv"
+  install_headless_job "sshnpd.sh"
+}
+
+install_headless_sshrvd() {
+  install_single_binary "sshrvd"
+}
+
+headless() {
+  case "$1" in
+    --help|'') usage; exit 0;;
+    sshnpd) install_headless_sshnpd;;
+    sshrvd) install_headless_sshrvd;;
+    *)
+      echo "Unknown headless job: $1";
+      usage;
+      exit 1;
+  esac
+}
+
+# TMUX SESSION #
+
+post_tmux_message() {
+  # Explain what needs to be edited
+  # Provide initial startup command
+  echo "tmux service installed, make sure to configure the service by editing $dest"
+  echo "Learn more in $script_dir/headless/README.md"
+  echo ""
+  echo "To start the service immediately:"
+  echo "  $command &"
+  echo ""
+  echo "Warning: sometimes tmux is not set to linger which will kill the tmux session on logout"
+  echo "Make sure your system is configured so that tmux sessions can linger, or disown the tmux process before logging out"
+}
```

Comment:
There are ways to automate this, but each solution has trade-offs, leave it to the user to pick and choose.